### ISSUE TITLE
Grafana fails for first start after a migration

### ISSUE THREAD

**Issue by astrobokonon** [LowellObservatory/DockerDev#4]
I *think* it's related to these lines in the docker-compose configuration:

https://github.com/LowellObservatory/DockerDev/blob/5ea27d3578891f7c9533e03b6c0f823d9a84efae/DCTStack/docker-compose.yml#L107

or 

https://github.com/LowellObservatory/DockerDev/blob/5ea27d3578891f7c9533e03b6c0f823d9a84efae/DCTStack/docker-compose.yml#L108

since I was seeing the following in the logs when trying to start it:

```
t=2018-12-18T00:16:56+0000 lvl=info msg="Starting Grafana" logger=server version=5.4.2 commit=d812109 branch=HEAD compiled=2018-12-13T12:49:23+0000
t=2018-12-18T00:16:56+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2018-12-18T00:16:56+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from command line" logger=settings arg="default.log.mode=console"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_DATA=/var/lib/grafana"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_LOGS=/var/log/grafana"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PLUGINS=/var/lib/grafana/plugins"
t=2018-12-18T00:16:56+0000 lvl=info msg="Config overridden from Environment variable" logger=settings var="GF_PATHS_PROVISIONING=/etc/grafana/provisioning"
t=2018-12-18T00:16:56+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2018-12-18T00:16:56+0000 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
t=2018-12-18T00:16:56+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2018-12-18T00:16:56+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2018-12-18T00:16:56+0000 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
t=2018-12-18T00:16:56+0000 lvl=info msg="App mode production" logger=settings
t=2018-12-18T00:16:56+0000 lvl=info msg="Initializing HTTPServer" logger=server
t=2018-12-18T00:16:56+0000 lvl=info msg="Initializing SqlStore" logger=server
t=2018-12-18T00:16:56+0000 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=sqlite3
t=2018-12-18T00:16:56+0000 lvl=info msg="Starting DB migration" logger=migrator
t=2018-12-18T00:16:56+0000 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: Migration failed err: unable to open database file"
```

**Comment by astrobokonon** [comment]
Since it's a DB migration error, I'm guessing it's grafana.db...but I'm not sure how to fix this at the moment.

**Comment by astrobokonon** [comment]
So the problem here is related to my bootstrapping of the main Grafana Dockerfile; since I'm building an image myself with the plugins specified by the env variable (```GF_INSTALL_PLUGINS```) the plugins get put into the ```/var/lib/grafana/plugins``` directory in the image.  But if I specify a bind mount from a local/host directory into /var/lib/grafana like I had hoped, the plugins directory obviously isn't there and the image copy gets blown away.

**Comment by astrobokonon** [comment]
Had to do a bit of a dance and reorganization, but I now have a working scheme that also makes the user data persist in the DockerData directory tree.

Closed by b9088006db0a78dc51327bc5ab0912fca12f7a45
